hw/top/defs.bzl

```diff
@@ -14,3 +14,44 @@ ALL_TOP_NAMES = [
     top.name
     for top in ALL_TOPS
 ]
+
+def _all_ip_names():
+    names = {}
+    for top in ALL_TOPS:
+        for ip in top.ips:
+            names[ip.name] = {}
```

Comment:
Would this be better as a comprehension? (also, maybe use an integer as the value, rather than empty dict?)
```
names = {ip.name: 1 for ip in top.ips for top in ALL_TOPS}
```